File: java/cdk8s-crd/src/main/java/imports/io/cert_manager/CertificateSpecKeystoresPkcs12Profile.java
package imports.io.cert_manager;

/**
 * Profile specifies the key and certificate encryption algorithms and the HMAC algorithm used to create the PKCS12 keystore.
 * <p>
 * Default value is <code>LegacyRC2</code> for backward compatibility.
 * If provided, allowed values are: <code>LegacyRC2</code>: Deprecated. Not supported by default in OpenSSL 3 or Java 20. <code>LegacyDES</code>: Less secure algorithm. Use this option for maximal compatibility. <code>Modern2023</code>: Secure algorithm. Use this option in case you have to always use secure algorithms (eg. because of company policy). Please note that the security of the algorithm is not that important in reality, because the unencrypted certificate and private key are also stored in the Secret.
 */
@javax.annotation.Generated(value = "jsii-pacmak/1.86.1 (build defb235)", date = "2024-01-04T09:05:26.259Z")
@software.amazon.jsii.Jsii(module = imports.io.cert_manager.$Module.class, fqn = "iocert-manager.CertificateSpecKeystoresPkcs12Profile")
public enum CertificateSpecKeystoresPkcs12Profile {
    /**
     * LegacyRC2.
     */
    LEGACY_RC2,
    /**
     * LegacyDES.
     */
    LEGACY_DES,
    /**
     * Modern2023.
     */
    MODERN2023,
}

File: go/cdk8s-crd/imports/certmanagerio/CertificateSpecKeystoresPkcs12Profile.go
package certmanagerio


// Profile specifies the key and certificate encryption algorithms and the HMAC algorithm used to create the PKCS12 keystore.
//
// Default value is `LegacyRC2` for backward compatibility.
//
// If provided, allowed values are:
// `LegacyRC2`: Deprecated. Not supported by default in OpenSSL 3 or Java 20.
// `LegacyDES`: Less secure algorithm. Use this option for maximal compatibility.
// `Modern2023`: Secure algorithm. Use this option in case you have to always use secure algorithms
// (e.g., because of company policy). Please note that the security of the algorithm is not that important
// in reality, because the unencrypted certificate and private key are also stored in the Secret.
type CertificateSpecKeystoresPkcs12Profile string

const (
	// LegacyRC2.
	CertificateSpecKeystoresPkcs12Profile_LEGACY_RC2 CertificateSpecKeystoresPkcs12Profile = "LEGACY_RC2"
	// LegacyDES.
	CertificateSpecKeystoresPkcs12Profile_LEGACY_DES CertificateSpecKeystoresPkcs12Profile = "LEGACY_DES"
	// Modern2023.
	CertificateSpecKeystoresPkcs12Profile_MODERN2023 CertificateSpecKeystoresPkcs12Profile = "MODERN2023"
)

